### PR TITLE
Quiet compiler

### PR DESCRIPTION
**Pull request by sethalves**
Quiet compiler

### CODE REVIEW

**Comment by sethalves** [comment]
consider adding ?w=1 to the end of the url while reviewing -- I fixed some line endings.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 804c1ecf8e2523545553b62da2d6587b883e63d0 available at: https://gist.github.com/8e3fb7256593f61017a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5551.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5551.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: fedbfe88815d25b11e68e0963162549a398d38c0 available at: https://gist.github.com/8e3fb7256593f61017a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5551.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5551.dmg

File: libraries/fbx/src/FBXReader.cpp [hunk 1]
```diff
@@ -1107,13 +1107,13 @@ ExtractedMesh extractMesh(const FBXNode& object, unsigned int& meshIndex) {
...
                 if (subdata.name == "Materials") {
                     materials = getIntVector(subdata);
-                } else if (subdata.name == "MappingInformationType") {
-                    if (subdata.properties.at(0) == "ByPolygon") {   
-                        isMaterialPerPolygon = true;
-                    } else {
-                        isMaterialPerPolygon = false;
-                    }
-                }
+                } // else if (subdata.name == "MappingInformationType") {
```
Comment (ZappoMan) [inline libraries/fbx/src/FBXReader.cpp]: ????

Comment (sethalves) [inline libraries/fbx/src/FBXReader.cpp]: isMaterialPerPolygon isn't used, so the entire block does nothing.  Should I get rid of it?

File: libraries/shared/src/PhysicsCollisionGroups.h [hunk 2]
```diff
@@ -44,7 +44,7 @@ const int16_t COLLISION_GROUP_OTHER_AVATAR = 1 << 6;
 const int16_t COLLISION_GROUP_MY_ATTACHMENT = 1 << 7;
 const int16_t COLLISION_GROUP_OTHER_ATTACHMENT = 1 << 8;
 // ...
-const int16_t COLLISION_GROUP_COLLISIONLESS = 1 << 15;
+const int16_t COLLISION_GROUP_COLLISIONLESS = 1 << 14;
```
Comment (howard-stearns) [inline libraries/shared/src/PhysicsCollisionGroups.h]: OK, but why? It's not obvious to me why this change is important/useful. Comment?

Comment (AndrewMeadows) [inline libraries/shared/src/PhysicsCollisionGroups.h]: The compiler was warning about shifting a 16-bit int all the way into a negative number.  By only shifting 14 bits the warning goes away.  I would have used **uint16_t** when I first wrote this, but Bullet really does use a 16-bit int and I wanted to agree with what they were using.

Comment (howard-stearns) [inline libraries/shared/src/PhysicsCollisionGroups.h]: Fair enough. (Could maybe add a comment // 14 to avoid changing sign on signed int used by Bullet. But not a blocker.)

File: libraries/fbx/src/FBXReader.cpp (hunk 1 above)
Comment (AndrewMeadows) [inline libraries/fbx/src/FBXReader.cpp]: In fact **isMaterialPerPolygon** IS used later, but used to set **isMultiMaterial** which is itself unused, so it does lead to a dead end.

Keep this logic I think.  Maybe someday we'll want to handle multi-material objects?  Of course, it would be nice to get rid of a warning about unused variable -- maybe set **isMaterialPerPolygon** equal to **isMultiMaterial** inside that check and add a comment about why that is being done (e.g. to remove a warning while still keeping the multi-material scan intact).

File: libraries/render-utils/src/Model.cpp [hunk 3]
```diff
@@ -1807,18 +1807,6 @@ void Model::segregateMeshGroups() {
...
-        bool translucentMesh = networkMesh.getTranslucentPartCount(mesh) == networkMesh.parts.size();
-        bool hasTangents = !mesh.tangents.isEmpty();
-        bool hasSpecular = mesh.hasSpecularTexture();
-        bool hasLightmap = mesh.hasEmissiveTexture();
-        bool isSkinned = state.clusterMatrices.size() > 1;
-        bool wireframe = isWireframe();
-        
-        if (wireframe) {
-            translucentMesh = hasTangents = hasSpecular = hasLightmap = isSkinned = false;
-        }
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/Model.cpp]: I think @samcake should look this over and decide if it should be removed, or if we are computing some values that are then incorrectly ignored and should actually be used.

File: libraries/fbx/src/FBXReader.cpp (hunk 1 above)
Comment (ZappoMan) [inline libraries/fbx/src/FBXReader.cpp]: I just don't like to see this commented out code here... it seems like this code is supposed to do something...

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1c8030472ae6a06f85ec0d9f71aa61278fd78d1b available at: https://gist.github.com/8e3fb7256593f61017a2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5551.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5551.dmg
